client/src/main/java/io/pravega/client/tables/Put.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.client.tables;
+
+import java.nio.ByteBuffer;
+import javax.annotation.Nullable;
+import lombok.Getter;
+import lombok.ToString;
+
+/**
+ * Updates the Value associated with a {@link TableKey} in a {@link KeyValueTable}.
+ */
+@ToString
+public final class Put extends TableEntryUpdate {
+    @Getter
+    private final Version version;
+
+    /**
+     * Creates a new instance of the {@link Put} class for an unconditional update.
+     *
+     * @param key   The {@link TableKey} to update.
+     * @param value The Value to associate with {@link TableKey}.
+     */
+    public Put(TableKey key, ByteBuffer value) {
+        this(key, value, null);
+    }
+
+    /**
+     * Creates a new instance of the {@link Put} class for a conditional update.
+     *
+     * @param key     The {@link TableKey} to update.
+     * @param value   The Value to associate with {@link TableKey}.
+     * @param version (Optional) The {@link Version} to condition the update on. This can be retrieved from
+     *                {@link TableEntry#getVersion()} or it may be {@link Version#NOT_EXISTS} or {@link Version#NO_VERSION}.
+     */
+    public Put(TableKey key, ByteBuffer value, @Nullable Version version) {
```

Comment:
Version can be both `null` and `NO_Version`. We should probably pick one that we want users to use.

Reply:
I clarified the Javadoc and defaulted this field to `Version.NO_VERSION`.  This makes it consistent with `Insert` which defaults it (internally) to `Version.NOT_EXISTS` as that is a conditional insert.